roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -72,6 +72,16 @@ def get_var(task_vars, *keys, **kwargs):
         raise OpenShiftCheckException("'{}' is undefined".format(".".join(map(str, keys))))
     return value
 
+def normalized_release(task_vars):
+    """Helper function to normalize the release format to dotted numbers"""
+    release = str(get_var(task_vars, "openshift_release"))
+    if release.startswith('v'):
+        release = release[1:]  # v3.3 => 3.3
+    return release
+
+def normalized_minor_release(task_vars):
```

Comment:
Changing `OpenShiftCheck.run`'s signatures brings a similar tradeoff than adding static methods.

In every subclass, calls to a module would become:

```python
class MyCheck(OpenShiftCheck):
    def run(self):
        self.module_executor("module_name", self.tmp, self.task_vars)
```

This loses the symmetry with action plugins, the run method would have a different signature.

If we keep diving that way, we could even assume that `self.tmp, and self.task_vars` are the common case and make a helper:

```python
class MyCheck(OpenShiftCheck):
    def run(self):
        self.execute_module("module_name")
```

Please do experiment and improve the "framework" as we progress and add more checks ;)